browser/profiles/delete_profile_helper_browsertest.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/profiles/delete_profile_helper.h"
+
+#include "base/files/file_path.h"
+#include "base/functional/callback_helpers.h"
+#include "base/run_loop.h"
+#include "base/test/bind.h"
+#include "brave/components/sync/service/brave_sync_service_impl.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/keep_alive/profile_keep_alive_waiter.h"
+#include "chrome/browser/profiles/profile_attributes_storage.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/profiles/profile_test_util.h"
+#include "chrome/browser/profiles/profiles_state.h"
+#include "chrome/browser/sync/sync_service_factory.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "content/public/test/browser_test.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace {
+const char kValidSyncCode[] =
+    "fringe digital begin feed equal output proof cheap "
+    "exotic ill sure question trial squirrel glove celery "
+    "awkward push jelly logic broccoli almost grocery drift";
+}
+
+// Inspired by src/chrome/browser/profiles/delete_profile_helper_browsertest.cc
+class DeleteProfileHelperBrowserTest : public InProcessBrowserTest {
+ public:
+  DeleteProfileHelperBrowserTest() = default;
+  ~DeleteProfileHelperBrowserTest() override = default;
+};
+
+IN_PROC_BROWSER_TEST_F(DeleteProfileHelperBrowserTest,
+                       SyncStoppedForDeletedProfile) {
+  // Create an additional profile.
+  ProfileManager* profile_manager = g_browser_process->profile_manager();
+  base::FilePath profile_path_to_delete =
+      profile_manager->GenerateNextProfileDirectoryPath();
+  Profile& profile_to_delete = profiles::testing::CreateProfileSync(
+      profile_manager, profile_path_to_delete);
+  EXPECT_TRUE(profile_manager->GetProfileAttributesStorage()
```

Comment:
This should be ASSERT because this is the precondition has to be fulfilled before testing the stuff we are interested in.

Reply:
yes, thanks, fixed at https://github.com/brave/brave-core/pull/24988/commits/8a5fa75b39c49f8ce53a6e31adf5200406b8d900#diff-3bcc7592f744cd83ffad9eaf185de467750fae6f5c2f17506f4b68a4f432840eR47